src/main/scala/DocGenerator.scala

```diff
@@ -67,130 +71,132 @@ object LatexDocGenerator extends DocGenerator {
-      tp.name.name + dumpTypeParams(tp.tparams) +
-      process(tp.typeBounds.lo, " >: ") +
-      process(tp.typeBounds.hi, " <: ") +
-      tp.contextBounds.collect { case e: Type.Name => s": ${e.name}" }.mkString(" ") +
-      tp.viewBounds.collect { case e: Type.Name => s"<% ${e.name}" }.mkString(" ")
-
-
-  }
-
-  def dumpTypeParams(tps: Seq[Type.Param]) = {
-    val params = tps.map(dumpTypeParam).mkString(", ")
-    if (params.nonEmpty) {
-      s"[$params]"
-    } else ""
-  }
 
   def processMethodsSummary(methods: Seq[Tree]): String = {
     s"""\\subsection{Method summary}{
-      \\begin{verse}
-        ${methods.collect { case e: Def => s"{\\bf def ${e.name.name}(${dumpMethodInputs(e)})}\\\\" }.mkString("\n")}
-      \\end{verse}
-      }"""
+        \\begin{verse}
+          ${methods.collect { case e: Def => s"{\\bf def ${e.name}(${dumpMethodInputs(e)})}\\\\" }.mkString("\n")}
+        \\end{verse}
+        }"""
   }
 
-  def dumpMethodInputs(e: Def): String = e.paramss.map(_.map(e => e.decltpe.asInstanceOf[Type.Name].name).mkString(", ")).mkString(")(")
+  def dumpMethodInputs(e: Def): String = e.paramss.map(_.map(e => dumpType(e.decltpe)).mkString(", ")).mkString(")(")
 
-  def commonIndex(elems: Seq[ {def name: Name}], link: (Name => String)): String = {
+  def commonIndex(elems: Seq[ {def name: String; def id: Seq[Tree]}]): String = {
     "\\begin{multicols}{2}\\noindent\n" +
-      elems.map(e => s"{${e.name.name}\\ref{${link(e.name)}}\\\\}").mkString("\n") + "\n" +
+      elems.map(e => s"{${hyperlink(e, Some(e.name))}\\\\}").mkString("\n") + "\n" +
       "\\end{multicols}"
   }
 
   def indexForMethods(index: Index): String = {
-    commonIndex(index.defs, e => e.name)
+    commonIndex(index.defs)
   }
 
   def indexForObjects(index: Index): String = {
-    commonIndex(index.objects, e => s"${e.name}_object")
+    commonIndex(index.objects)
   }
 
   def indexForClasses(index: Index): String = {
-    commonIndex(index.classes, e => s"${e.name}_class")
+    commonIndex(index.classes)
   }
 
   def indexForTraits(index: Index): String = {
-    commonIndex(index.traits, e => s"${e.name}_trait")
+    commonIndex(index.traits)
   }
 
 
-  def dumpSignature(e: Def) = e.paramss.map(_.map(e => e.name + " : " + e.decltpe.asInstanceOf[Type.Name].name).mkString(", ")).mkString(")(")
+  def dumpSignature(e: Def) = {
+    e.paramss.map(_.map(e => e.name + " : " + dumpType(e.decltpe)).mkString(", ")).mkString(")(")
+  }
+
+  def dumpTypeParams(tps: Seq[Type.Param]) = {
+    val params = tps.map(dumpType).mkString(", ")
+    if (params.nonEmpty) {
+      s"[$params]"
+    } else ""
+  }
+
+  def dumpType(tpe: Type): String = {
+
+    tpe match {
+      case e: Type.Name => hyperlink(e, Some(e.name))
+      case e: Type.Apply => dumpType(e.tpe) + "[" + e.args.map(dumpType).mkString(", ") + "]"
+      case e: Type.Compound => e.tpes.map(dumpType).mkString("with ")
```

Comment:
Will you end up with something like `Xwith Ywith Z`, since you do `mkString("with ")` ?